shared/src/DebugBuildConfiguration.cs

```diff
@@ -3,6 +3,7 @@
 
 using System.Runtime.CompilerServices;
 
+[assembly: InternalsVisibleTo("Microsoft.Azure.Devices.Client")]
```

Comment:
ahh, this will be required!
we'll also need to add the public key, since both of these are strong-name signed: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-5.0

Reply:
another option is to add a compile include for these "common" files to the respective assembly (device client nuget), instead of putting them in the shared library. what do you think?

Reply:
No, I truly **loathe** this approach and would like to avoid it at all costs. If we planned to do that we need to create a shared project